vsintegration/src/FSharp.Editor/QuickInfoProvider.fs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Composition
+open System.Collections.Concurrent
+open System.Collections.Generic
+open System.Collections.Immutable
+open System.Threading
+open System.Threading.Tasks
+open System.Linq
+open System.Runtime.CompilerServices
+open System.Windows
+open System.Windows.Controls
+open System.Windows.Media
+
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.Completion
+open Microsoft.CodeAnalysis.Classification
+open Microsoft.CodeAnalysis.Editor
+open Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.QuickInfo
+open Microsoft.CodeAnalysis.Editor.Shared.Utilities
+open Microsoft.CodeAnalysis.Formatting
+open Microsoft.CodeAnalysis.Host.Mef
+open Microsoft.CodeAnalysis.Options
+open Microsoft.CodeAnalysis.Text
+
+open Microsoft.VisualStudio.FSharp.LanguageService
+open Microsoft.VisualStudio.Text
+open Microsoft.VisualStudio.Text.Tagging
+open Microsoft.VisualStudio.Shell
+open Microsoft.VisualStudio.Shell.Interop
+
+open Microsoft.FSharp.Compiler
+open Microsoft.FSharp.Compiler.Parser
+open Microsoft.FSharp.Compiler.Range
+open Microsoft.FSharp.Compiler.SourceCodeServices
+
+// FSROSLYNTODO: with the merge of the below PR, the QuickInfo API should be changed
+// to allow for a more flexible syntax for defining the content of the tooltip.
+// The below interface should be discarded then or updated accourdingly.
+// https://github.com/dotnet/roslyn/pull/13623
+type internal FSharpDeferredQuickInfoContent(content: string) =
+    interface IDeferredQuickInfoContent with
+        override this.Create() : FrameworkElement =
+            let label = new Label()
+            label.Content <- content
+            label.Foreground <- SolidColorBrush(Colors.White)
+            upcast label
+
+[<Shared>]
+[<ExportQuickInfoProvider(PredefinedQuickInfoProviderNames.Semantic, FSharpCommonConstants.FSharpLanguageName)>]
+type internal FSharpQuickInfoProvider [<ImportingConstructor>] (serviceProvider: SVsServiceProvider) =
```

Comment:
Yes, C# provider imports other dependencies, but I don't think it's the problem. We successfully import `SVsServiceProvider` in several places. I don't know much about MEF either :(